src/clients/UBAClient/UBAClientBase.ts

```diff
@@ -146,12 +148,36 @@ export abstract class BaseUBAClient {
    * @param toBlock         Optional upper bound of the search range. Defaults to the latest queried block.
    */
   public getFlows(chainId: number, tokenSymbol: string, fromBlock?: number, toBlock?: number): UbaFlow[] {
+    return this.getFlowsWithClosingBalances(chainId, tokenSymbol, fromBlock, toBlock).map(({ flow }) => flow);
+  }
+
+  /**
+   * Construct the ordered sequence of SpokePool flows between two blocks. This function returns the flows with closing balances.
+   * @note Assumptions:
+   * @note Deposits, Fills and RefundRequests have been pre-verified by the SpokePool contract or SpokePoolClient, i.e.:
+   * @note - Deposit events contain valid information.
+   * @note - Fill events correspond to valid deposits.
+   * @note - RefundRequest events correspond to valid fills.
+   * @note In order to provide up-to-date prices, UBA functionality may want to follow close to "latest" and so may still
+   * @note be exposed to finality risk. Additional verification that can only be performed within the UBA context:
+   * @note - Only the first instance of a partial fill for a deposit is accepted. The total deposit amount is taken, and
+   * @note   subsequent partial, complete or slow fills are disregarded.
+   * @param spokePoolClient SpokePoolClient instance for this chain.
+   * @param fromBlock       Optional lower bound of the search range. Defaults to the SpokePool deployment block.
+   * @param toBlock         Optional upper bound of the search range. Defaults to the latest queried block.
+   * @returns The flows with closing balances for the given token on the given chainId between the given block numbers
+   */
+  public getFlowsWithClosingBalances(
```

Comment:
nit: rename to `getModifiedFlows`?

Reply:
Done https://github.com/across-protocol/sdk-v2/pull/278/commits/8ef9f0f7ce7d2dcd18d09240d793d4bf56d2250e